python/tests/test_writer.py

```diff
@@ -496,3 +496,218 @@ def test_writer_with_options(tmp_path: pathlib.Path):
     )
 
     assert table == data
+
+
+@pytest.mark.parametrize(
+    "cases_data",
+    [
```

Comment:
Sorry this ended up quite long!

What if instead of varying the data, you used the same table for all of them but varied which column in the table was the partition column. See, [this test as an example](https://github.com/delta-io/delta-rs/blob/d984ccfdb5ae3d8ddfc4b6611c6c9ad599541cc9/python/tests/test_writer.py#L153-L182).

Then maybe you can parametrize by 

```python
@pytest.mark.parametrize(
    "cases",
    [("int64", [("int64", "=", "1")], [("int64", ">", "1")]),
     ("bool", [("bool", "=", "true")], [("bool", "!=", "true")]),
     ...
    ]
)
```

If it ends up not being possible to make more succinct, that fine. But it seems worth a try 😄

Reply:
Updates